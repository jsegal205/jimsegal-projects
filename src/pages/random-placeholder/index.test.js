import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import {
  render,
  cleanup,
  getByTestId,
  queryByTestId,
  fireEvent,
} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { toMatchDiffSnapshot } from "snapshot-diff";

expect.extend({ toMatchDiffSnapshot });
import RandomPlaceholder from "./index";

jest.mock("./sites", () => {
  return {
    key: {
      apiLink: "http://apiLink/",
      delimiter: "x",
    },
  };
});

afterEach(cleanup);

describe("Random Placeholder Component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Router>
        <RandomPlaceholder />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  describe("input logic", () => {
    it("defaults inputs to empty", () => {
      const { container } = render(
        <Router>
          <RandomPlaceholder />
        </Router>
      );

      const heightInput = getByTestId(container, "placeholder-height");
      const widthInput = getByTestId(container, "placeholder-width");

      expect(heightInput.value).toBe("");
      expect(widthInput.value).toBe("");
      expect(
        queryByTestId(container, "placeholder-image")
      ).not.toBeInTheDocument();
      expect(
        queryByTestId(container, "placeholder-error")
      ).not.toBeInTheDocument();
    });

    describe("when inputs are blank", () => {
      it("shows error messaging", () => {
        const { container, asFragment } = render(
          <Router>
            <RandomPlaceholder />
          </Router>
        );

        const initialRender = asFragment();
        const btn = getByTestId(container, "placeholder-submit");
        fireEvent.click(btn);

        expect(initialRender).toMatchDiffSnapshot(asFragment());
      });
    });

    describe("when inputs are filled", () => {
      it("shows image", () => {
        const { container, asFragment } = render(
          <Router>
            <RandomPlaceholder />
          </Router>
        );

        const initialRender = asFragment();
        const heightInput = getByTestId(container, "placeholder-height");
        const widthInput = getByTestId(container, "placeholder-width");

        fireEvent.change(heightInput, { target: { value: "1" } });
        fireEvent.change(widthInput, { target: { value: "1" } });

        const btn = getByTestId(container, "placeholder-submit");
        fireEvent.click(btn);

        expect(initialRender).toMatchDiffSnapshot(asFragment());
      });
    });
  });
});
