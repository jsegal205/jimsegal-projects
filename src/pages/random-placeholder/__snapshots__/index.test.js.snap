// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Placeholder Component input logic when inputs are blank shows error messaging 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -20,11 +20,11 @@
            </div>
            <div
              class=\\"col two-third\\"
            >
              <input
-               class=\\"\\"
+               class=\\"placeholder-input-error\\"
                data-testid=\\"placeholder-height\\"
                type=\\"number\\"
                value=\\"\\"
              />
              <label>
@@ -44,11 +44,11 @@
            </div>
            <div
              class=\\"col two-third\\"
            >
              <input
-               class=\\"\\"
+               class=\\"placeholder-input-error\\"
                data-testid=\\"placeholder-width\\"
                type=\\"number\\"
                value=\\"\\"
              />
              <label>
@@ -61,8 +61,14 @@
          class=\\"placeholder-btn\\"
          data-testid=\\"placeholder-submit\\"
        >
          Get Random Placeholder Image
        </button>
+       <div
+         class=\\"placeholder-error\\"
+         data-testid=\\"placeholder-error\\"
+       >
+         Please enter height and width
+       </div>
      </div>
    </section>
  </DocumentFragment>"
`;

exports[`Random Placeholder Component input logic when inputs are filled shows image 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -23,11 +23,11 @@
            >
              <input
                class=\\"\\"
                data-testid=\\"placeholder-height\\"
                type=\\"number\\"
-               value=\\"\\"
+               value=\\"1\\"
              />
              <label>
                (px)
              </label>
            </div>
@@ -47,11 +47,11 @@
            >
              <input
                class=\\"\\"
                data-testid=\\"placeholder-width\\"
                type=\\"number\\"
-               value=\\"\\"
+               value=\\"1\\"
              />
              <label>
                (px)
              </label>
            </div>
@@ -61,8 +61,25 @@
          class=\\"placeholder-btn\\"
          data-testid=\\"placeholder-submit\\"
        >
          Get Random Placeholder Image
        </button>
+     </div>
+     <div
+       class=\\"placeholder\\"
+       data-testid=\\"placeholder-image\\"
+     >
+       This placeholder by 
+       <a
+         href=\\"key\\"
+         rel=\\"noopener noreferrer\\"
+         target=\\"_blank\\"
+       >
+         key
+       </a>
+       <img
+         alt=\\"placeholder sourced from key\\"
+         src=\\"http://apiLink/1x1\\"
+       />
      </div>
    </section>
  </DocumentFragment>"
`;

exports[`Random Placeholder Component renders correctly 1`] = `
<div>
  <section>
    Overwhelmed with the placeholder image sites for your designs? Let me aggregrate all the favorites!
    <div
      class="dimensions"
    >
      <h3>
        Enter dimensions
      </h3>
      <div>
        <div
          class="input"
        >
          <div
            class="col one-third"
          >
            <label>
              Height:
            </label>
          </div>
          <div
            class="col two-third"
          >
            <input
              class=""
              data-testid="placeholder-height"
              type="number"
              value=""
            />
            <label>
              (px)
            </label>
          </div>
        </div>
        <div
          class="input"
        >
          <div
            class="col one-third"
          >
            <label>
              Width:
            </label>
          </div>
          <div
            class="col two-third"
          >
            <input
              class=""
              data-testid="placeholder-width"
              type="number"
              value=""
            />
            <label>
              (px)
            </label>
          </div>
        </div>
      </div>
      <button
        class="placeholder-btn"
        data-testid="placeholder-submit"
      >
        Get Random Placeholder Image
      </button>
    </div>
  </section>
</div>
`;
